src/UI/Areas/Identity/Pages/Account/ExternalLogin.cshtml.cs

```diff
@@ -124,7 +128,7 @@ public override IActionResult OnPost(string provider, string returnUrl = null)
 
             if (ModelState.IsValid)
             {
-                var user = new TUser { UserName = Input.Email, Email = Input.Email };
+                var user = _userInstanceFactory.CreateUser(email: Input.Email, userName: Input.Email);
```

Comment:
```csharp
var user = new TUser();
await _userStore.SetEmailAsync(input.Email);
await _userStore.SetUserNameAsync(input.Email);
_userManager.CreateAsync(user);
```

Reply:
That will work, with the tradeoff of requiring overriding multiple views required whenever they want to add required data, but if that's already required I guess that's fine.

Reply:
This is sorta breaking the abstractions though, the whole point is you aren't supposed to use the stores directly, only the manager...

Reply:
The password hash null check is unavoidable since the only workaround would be calling AddPassword with a null password and checking if that fails, which is super gross.

Reply:
You'd have to do a cast/check to make sure the user store even supports Email as well going this route...

Reply:
I think in the end, the core abstraction that the UI needs to introduce around custom users, is around the registration model and creating that initial instance.

Basically what's needed is a way to plug in a user data schema (property name + data type), and generate some UI around that data by default, and then create a TUser instance with that data, and then pass that into Identity Core.

We can file an issue to track this separately, I guess for preview 2 we can just hack something in that works for now and revisit this in rc if needed

Reply:
The user manager doesn't have the functionality. If you are OK adding it, we can just add it there. This can take multiple forms:
```csharp
_userManager.Create(userName, email)
```
```
_userManager.SetUserNameAsync(user, userName, updateUser:false)
_userManager.SetEmailAsync(user, email, updateUser:false)
```